### ISSUE TITLE
The "output-module-groups" CLI command is not working on terragrunt >= "0.56.4"

### ISSUE THREAD

**Issue by rodrigorfk** [gruntwork-io/terragrunt#3063]
**Describe the bug**
The "output-module-groups" CLI command stopped working on recent terragrunt versions and the command is failing with the following error:
```
ERRO[0000] lstat : no such file or directory
ERRO[0000] Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```
<img width="918" alt="image" src="https://github.com/gruntwork-io/terragrunt/assets/1995033/b579123f-8be4-44fe-8721-6f16144a7946">

**To Reproduce**
```
# install the latest terragrunt version

git clone https://github.com/gruntwork-io/terragrunt-infrastructure-live-example.git
cd terragrunt-infrastructure-live-example/prod
terragrunt output-module-groups apply
```

**Expected behavior**
I would expect a json output as generated on terragrunt <= 0.56.3 as following:
```json
{
  "Group 1": [
    "/Users/rodrigokuntzer/dev/temp/terragrunt-infrastructure-live-example/prod/us-east-1/prod/mysql",
    "/Users/rodrigokuntzer/dev/temp/terragrunt-infrastructure-live-example/prod/us-east-1/prod/webserver-cluster"
  ]
}
```

**Nice to have**
- [x] Terminal output
- [x] Screenshots

**Versions**
- Terragrunt version: 0.57.1
- Terraform version: 1.5.5
- Environment details (Ubuntu 20.04, Windows 10, etc.): MacOS 14.3.1

**Comment by rodrigorfk** [comment]
Hi @levkohimins, I hope you are doing well, perhaps you can help with this issue? looks like that #3001, which is part of the v0.56.4 has introduced the regression here. I have just tested with the latest v0.58.2 version and the problem remains.

**Comment by levkohimins** [comment]
Hi @rodrigorfk, Thank you for your kind words. I will look into this issue soon, most likely next week.

**Comment by levkohimins** [comment]
Hi @rodrigorfk, sorry for delay. I am trying to reproduce the issue with the latest Terragrunt version [v0.58.7](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.58.7) but it works without issues. Could you please check it out?

**Comment by rodrigorfk** [comment]
Hi @levkohimins , thanks for checking it, I just tried with version `v0.58.7` and from my perspective the issue remains there, one interesting finding is, `output-module-groups` does work if you omit the second argument (`<command`), meaning `apply` or `destroy`, which is supposed to fallback to `apply`, however if you pass that second argument, the command, as I have reported originally, does not work anymore:

![image](https://github.com/gruntwork-io/terragrunt/assets/1995033/1f7a7bd0-8d9d-42f9-8154-aa652b3c2253)

**Comment by levkohimins** [comment]
Hi @rodrigorfk, since I can't reproduce this issue in my environment, may I ask to run the command with
`TERRAGRUNT_LOG_LEVEL=debug` and show me with the full log?

**Comment by levkohimins** [comment]
Hi @rodrigorfk, I was able to repro the issue. I'll let you know when I fix the issue. Thanks.

**Comment by levkohimins** [comment]
Resolved in [v0.58.9](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.58.9) release.

**Comment by rodrigorfk** [comment]
Thank you very much @levkohimins, I can confirm the issue as being fixed.
